### ISSUE TITLE
Record Project Video feature

### ISSUE THREAD

**Issue by apple502j** [LLK/scratch-gui#5699]
Record Project Video feature in Scratch 2.0 recorded the project's stage as a FLV. Using [MediaRecorder](https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder) it is possible to record the stage (and possibly the sound): I made one [here](https://github.com/apple502j/scratch-gui/tree/stage-record-again) that records the stage (without sound) as WebM.

Options:
- Include Sound From Project: Possible, but requires refactoring scratch-audio so that we can inject `createMediaStreamDestination`
- Include Sound From Microphone: Possible (see above)
- Show Mouse Pointer: Not possible AFAIK
- Record Entire Editor: Not possible AFAIK
- Record at Highest Quality: Maybe?

Some forum posts:
- https://scratch.mit.edu/discuss/topic/394121/?page=1#post-3913157
- https://scratch.mit.edu/discuss/topic/336734/?page=1#post-3405115

**Comment by BoomerScratch** [comment]
Sounds awesome! 😃

**Comment by easrng** [comment]
> * Show Mouse Pointer: Not possible AFAIK

That should be possible. You'd need to ex.
```pseudocode
let pointerPositions;
on recording start{
  startTime=new Date().valueOf()
  pointerPositions = []
  on mouse move(x, y){
    pointerPositions.push([x,y,new Date().valueOf()-startTime])
  }
}
on recording end{
  remove mouse move handler
  add pointers to video(pointerPositions)
  pointerPositions=[]
}
```
Hmm. It might need a library to manipulate the frames though.

**Comment by easrng** [comment]
> * Record Entire Editor: Not possible AFAIK

Yeah, not really possible. I mean possibly something could be hacked together with dom2image or similar, but it's not really feasible. Actually, it may be possible to use getDisplayMedia for this.
